### ISSUE TITLE
asp.net core bin folder on publish to group all binaries

### ISSUE THREAD

**Issue by valoghia** [dotnet/aspnetcore#47982]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

When publishing an ASP.net Core app we get a folder with 
lots of folders and dll files (+ folders for each culture),
appsettings.json, web.config

### Describe the solution you'd like

it would be much better to have a bin folder for all the dll files and rest non wwwroot folders

### Additional context

Might not be a big issue for users of automatic/azure deployments, but for users of shared hosting and unperfect file managers,
where we have to pick and delete files before unzipping the deployment zip it makes things harder.

On each deployment I have to delete all files, except:
 `.well-known`, my custom `App_Data` folder(s), wwwroot sometimes, `appsettings.json` sometimes.

And when using `mvcBuilder.AddRazorRuntimeCompilation()` there's also the Views folder.

So basically I need to do a whole lot of checkbox ticking and if all the dlls/runtimes folders and files would be in a `bin` folder it would make things a lot easier.

**Comment by davidfowl** [comment]
> but for users of shared hosting and unperfect file managers,
where we have to pick and delete files before unzipping the deployment zip it makes things harder.

Can you elaborate why you have to delete files?

**Comment by valoghia** [comment]
@davidfowl when I need to deploy a new version of the website, 
I zip the new published files, 
I upload the zip
delete existing dll files, but I have to leave some of the storage (App_Data like) folders and .well-known (let's encrypt related), and uploaded zip
after I unzip
(overwrite doesn't work in my case, file manager error)

**Comment by davidfowl** [comment]
I can’t see us bringing back the bin folder to make uploading to shared hosters easier (ftp?). If your shared hosted supports web deploy; then it’s possible to automate the delete and push (it can diff what’s local and what’s remote).  

It might be possible to configure the settings in a way that makes this work, but it’s a little tricky if you are using IIS

**Comment by valoghia** [comment]
@davidfowl you could make it as setting that is turned `off` by default.

**Comment by davidfowl** [comment]
There would need to be a large demand for us to investigate adding this feature. We can leave this issue open and see how big of a pain point it is in general for customers. I assume you are using a traditional windows shared hoster? If not, can you share more details on your deployment process?

**Comment by valoghia** [comment]
@davidfowl I'm using smarterasp.net,
I'm deploying by uploading a zip file, ftp (filezilla) is usually a lot slower comparing to uploading a single zip;
and I might still need to select and delete old files
